### ISSUE TITLE
Provide an official Firefox extension

### ISSUE THREAD

**Issue by ingorichter** [tabwrangler/tabwrangler#185]
I'm using a modified unsigned version of the extension in debug mode for some time now. It works fine so far, and I'd like to make it an official Firefox extension.

- [x] Create an XPI package
- [x] Sign the extension. Do we need keys for signing?
- [x] Distribute to addons.mozilla.org
- [x] Use alternative favicon loading instead of Chrome-specific `"chrome://favicon/"`

This https://github.com/tabwrangler/tabwrangler/issues/168 might need to happen before publishing it to the Firefox extensions marketplace

**Comment by ssorallen** [comment]
There's no need to create an XPI package; Firefox accepts .zip files just like Chrome. I changed the .zip file to pack everything at the root to work with Firefox's uploads: 96851863e002b82b539886d306865ffbcf24ddf6

**Comment by ssorallen** [comment]
No need for signing either; that is done by the Firefox Developer Hub.

**Comment by ingorichter** [comment]
Hm, did you see any warnings regarding the manifest? I don't remember the exact warnings, but there was something wrong. Anyway, if it works for you that's great. I saw some visual glitches on FF. There is some testing required before it can be published for FF. And we need some artwork and screenshots. I can take screenshots

**Comment by ssorallen** [comment]
@ingorichter Yes I saw a problem about the manifest too. The problem was the .zip file nested everything inside a directory called `dist/`. With commit https://github.com/tabwrangler/tabwrangler/commit/96851863e002b82b539886d306865ffbcf24ddf6 the contents of the .zip are all at the root.

**Comment by ssorallen** [comment]
Firefox does not like the Chrome-specific favicon permission. That will either need to be removed entirely, or the build step will need a non-Chrome version. I'm not a fan of either direction, but I lean toward the former because managing two versions of the extension sounds annoying.

<img width="810" alt="screen shot 2018-04-04 at 6 15 42 pm" src="https://user-images.githubusercontent.com/29612/38302197-6a425c00-3834-11e8-86b7-d6cc308a7e1d.png">

Text content of the image:

> Reading manifest: Error processing permissions.0: Value "chrome://favicon/" must either: must either [must either [be one of ["clipboardRead", "clipboardWrite", "geolocation", "idle", "notifications"], be one of ["bookmarks"], be one of ["find"], be one of ["history"], be one of ["activeTab", "tabs"], be one of ["browserSettings"], be one of ["cookies"], be one of ["topSites"], be one of ["webNavigation"], or be one of ["webRequest", "webRequestBlocking"]], be one of ["alarms", "mozillaAddons", "storage", "unlimitedStorage"], be one of ["browsingData"], be one of ["devtools"], be one of ["identity"], be one of ["menus", "contextMenus"], be one of ["pkcs11"], be one of ["geckoProfiler"], be one of ["sessions"], be one of ["contextualIdentities"], be one of ["downloads", "downloads.open"], be one of ["management"], be one of ["privacy"], be one of ["proxy"], be one of ["nativeMessaging"], be one of ["theme"], or match the pattern /^experiments(\.\w+)+$/], or must either [be one of ["<all_urls>"], match the pattern /^(https?|wss?|file|ftp|\*):\/\/(\*|\*\.[^*/]+|[^*/]+)\/.*$/, or match the pattern /^file:\/\/\/.*$/]
> 
> Reading manifest: Error processing commands.wrangle-current-tab.suggested-key: An unexpected property was found in the WebExtension manifest.
> 
> Reading manifest: Error processing options_page: An unexpected property was found in the WebExtension manifest.

**Comment by Marcelovk** [comment]
I added a gulp file with a new task for creating a signed XPI here: https://github.com/tabwrangler/tabwrangler/pull/222. 

You only need to get a set of keys in addons.mozilla.org

The issue of loading a zip in Firefox is that it will be only avaliable until the next restart, because the addon is not signed. 

I am attaching here the xpi generated with my keys, please remove it if is not desirable. 

[tab_wrangler-5.11.0-an+fx.xpi.zip](https://github.com/tabwrangler/tabwrangler/files/1923130/tab_wrangler-5.11.0-an.fx.xpi.zip)

**Comment by ssorallen** [comment]
We are live 🎉🎉🎉 with Tab Wrangler for Firefox:

https://addons.mozilla.org/en-US/firefox/addon/tabwrangler/

**Comment by ingorichter** [comment]
Awesome 👏 @ssorallen
